File: Programmers/코딩테스트 고득점 kit/해시/폰켓몬.py
# https://school.programmers.co.kr/learn/courses/30/lessons/1845

def solution(nums):
    answer = 0
    pokemon = {}
    for i in range(len(nums)):
        pokemon[nums[i]] = pokemon.get(nums[i], 0) + 1
    if len(pokemon) >= len(nums) / 2:
        answer = len(nums) // 2
    else:
        answer = len(pokemon)
    return answer

File: Programmers/코딩테스트 고득점 kit/해시/java/폰켓몬.java
// https://school.programmers.co.kr/learn/courses/30/lessons/1845

import java.util.*;

class Solution {
    public int solution(int[] nums) {
        int answer = nums.length / 2;
        
        HashSet<Integer> set = new HashSet<>();
        for (int num : nums) {
            set.add(num);
        }
        
        if (answer >= set.size()) {
            answer = set.size();
        }
        
        return answer;
    }
}